### DISCUSSION TITLE
failedReason: "job stalled more than allowable limit". Can I increase this limit somehow?

### DISCUSSION THREAD

**Discussion by Arun-N** [OptimalBits/bull#1983] [Q&A]
Hi! What exactly is this limit? Is it based on time or some other factor? Currently I have a job that makes multiple API requests and takes a while to complete. Not sure if it's because of that, but this particular job is failing quite a lot with "job stalled more than allowable limit" as the reason.

I can't optimize the job code any further (to take less time) since it's dependent on the API call latency, which can take on an average, around 10-20 seconds, or more depending on traffic. So instead, can I increase the limit that's being used somehow?

Thanks.

**Comment by manast** [comment]
The error is emitted because your job stalled and default value is 1 for the ```maxStalledCount``` option: https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queue
You can increase it so that the job is re-run after it got stalled, however it is strange that your jobs stalls to begin with, either your processor function is not correctly written or it is consuming a lot of CPU.
